### PR TITLE
[backport][attributetable] Fix sorting with geometry expression Manual Backport #57624 

### PR DESCRIPTION
**Pull request by elpaso**
Manual backport of: https://github.com/qgis/QGIS/pull/57624

[attributetable] Fix sorting with geometry expression
Fix https://github.com/qgis/QGIS/issues/57338

### CODE REVIEW

File: src/gui/attributetable/qgsattributetablemodel.cpp [hunk 1]
```diff
@@ -42,9 +42,7 @@
 #include <limits>
 
 QgsAttributeTableModel::QgsAttributeTableModel( QgsVectorLayerCache *layerCache, QObject *parent )
-  : QAbstractTableModel( parent )
-  , mLayer( layerCache->layer() )
-  , mLayerCache( layerCache )
+  : QAbstractTableModel( parent ), mLayer( layerCache->layer() ), mLayerCache( layerCache )
```
Comment (nyalldawson) [inline src/gui/attributetable/qgsattributetablemodel.cpp:45]:
```suggestion
  : QAbstractTableModel( parent )
  , mLayer( layerCache->layer() )
  , mLayerCache( layerCache )
```

Comment (nyalldawson) [reply src/gui/attributetable/qgsattributetablemodel.cpp:45]: The formatting updates through this PR are rather annoying...

File: src/gui/attributetable/qgsattributetablemodel.cpp [hunk 2]
```diff
@@ -673,19 +669,7 @@ QVariant QgsAttributeTableModel::headerData( int section, Qt::Orientation orient
...
-         && role != FeatureIdRole
-         && role != FieldIndexRole
-         && role != Qt::BackgroundRole
-         && role != Qt::ForegroundRole
-         && role != Qt::DecorationRole
-         && role != Qt::FontRole
-         && role < SortRole
-       )
-     )
+       ( role != Qt::TextAlignmentRole && role != Qt::DisplayRole && role != Qt::ToolTipRole && role != Qt::EditRole && role != FeatureIdRole && role != FieldIndexRole && role != Qt::BackgroundRole && role != Qt::ForegroundRole && role != Qt::DecorationRole && role != Qt::FontRole && role < SortRole ) )
```
Comment (nyalldawson) [inline src/gui/attributetable/qgsattributetablemodel.cpp:672]: revert? this makes it much less readable

Comment (elpaso) [reply src/gui/attributetable/qgsattributetablemodel.cpp:672]: TBH I don't know which one of my editors made this unwanted mess ... I suspect vscode that I used to fix conflicts.

File: src/gui/attributetable/qgsattributetablemodel.cpp (hunk 1 above)
Comment (elpaso) [reply src/gui/attributetable/qgsattributetablemodel.cpp]: > The formatting updates through this PR are rather annoying...

I've redone the PR, no more formatting mess.

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**